src/main/scala/org/broadinstitute/dsde/workbench/avram/api/BaseEndpoint.scala

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.dsde.workbench.avram.api
+
+import javax.servlet.http.HttpServletRequest
+import org.broadinstitute.dsde.workbench.avram.Avram
+import org.broadinstitute.dsde.workbench.avram.dataaccess.SamUserInfoResponse
+import org.broadinstitute.dsde.workbench.avram.util.ErrorResponse
+
+trait BaseEndpoint {
+  private val samDao = Avram.samDao
+
+  def handleAuthenticatedRequest[T]
+      (request: HttpServletRequest)
+      (f: SamUserInfoResponse => Either[ErrorResponse, T]): T = {
+    unsafeRun {
+      for {
+        userInfo <- extractUserInfo(request)
+        result <- f(userInfo)
+      } yield result
+    }
+  }
+
+  private def unsafeRun[T](f: => Either[ErrorResponse, T]): T = {
+    f.fold(e => throw e.exception, identity)
+  }
+
+  private def extractUserInfo(r: HttpServletRequest): Either[ErrorResponse, SamUserInfoResponse] = {
+    for {
+      token <- extractBearerToken(r)
+      userInfo <- samDao.getUserStatus(token)
+    } yield userInfo
+  }
+
+  private def extractBearerToken(r: HttpServletRequest): Either[ErrorResponse, String] = {
+    val token = for {
+      value <- Option(r.getHeader("Authorization"))
+      tokenMatch <- """(?i)bearer (.*)""".r.findPrefixMatchOf(value)
```

Comment:
It may be good to define the pattern in a class-level `val` to avoid recompiling it each time:
```
val bearerPattern = """(?i)bearer (.*)""".r
```